static/js/alert_words_ui.js

```diff
@@ -31,55 +31,60 @@ function update_alert_word_status(status_text, is_error) {
     alert_word_status.show();
 }
 
-function update_alert_words() {
-    var words = _.map($('.alert-word-item'), function (e) {
-        return $(e).data('word').toString();
-    });
-    words = _.filter(words, function (word) {
-        return word !== "";
-    });
-    channel.post({
-        url: '/json/users/me/alert_words',
-        idempotent: true,
-        data: {alert_words: JSON.stringify(words)}});
-}
-
-function add_alert_word(word, event) {
-    if ($.trim(word) === '') {
-        update_alert_word_status(i18n.t("Alert words can't be empty!"), true);
+function add_alert_word(alert_word) {
+    if ($.trim(alert_word) === '') {
+        update_alert_word_status(i18n.t("Alert word can't be empty!"), true);
         return;
     }
-    var final_li = templates.render('alert_word_settings_item', {word: word, editing: false});
 
-    var li = $(event.target).parents('li');
-    li.replaceWith(final_li);
+    var words = page_params.alert_words;
+    words.push(alert_word);
```

Comment:
I don't think that this change is needed with the proper flow, unless I'm missing something?